cpp/src/arrow/array/builder_nested.h

```diff
@@ -115,8 +138,16 @@ class ARROW_EXPORT VarLengthListLikeBuilder : public ArrayBuilder {
     return Status::OK();
   }
 
+  /// \brief Append an empty list slot
+  ///
+  /// \post Another call to Append* or Finish should be made before appending to
+  /// the values builder to ensure list slot remains empty
```

Comment:
Note that this was already an unwritten contract when there were only regular list builders.